examples/oneapi/cpp/source/example_util/csr_helpers.hpp

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+* Copyright 2023 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+#include "oneapi/dal/table/csr.hpp"
+#include <cmath>
+
+namespace dal = oneapi::dal;
+using oneapi::dal::detail::empty_delete;
+
+/// Converts a homogen table to one-based CSR table.
+///
+/// @param[in] data     an input homogen table
+inline const dal::csr_table convert_to_csr(const dal::table& data) {
+    std::int64_t non_zero_count = 0;
+    dal::row_accessor<const float> accessor{ data };
+    const auto data_ptr = accessor.pull({ 0, -1 });
+    const std::int64_t column_count = data.get_column_count();
+    const std::int64_t row_count = data.get_row_count();
+
+    for (std::int64_t i = 0; i < data_ptr.get_count(); i++) {
+        if (std::fabs(data_ptr[i]) > std::numeric_limits<float>::epsilon()) {
```

Comment:
```suggestion
        if (std::fabs(data_ptr[i]) > std::numeric_limits<Type>::epsilon()) {
```

Reply:
Or what is even better:
```suggestion
        num_zero_count += (std::fabs(data_ptr[i]) > std::numeric_limits<Type>::epsilon());
```

Presumably it should result in faster code